src/async/timing.ts

```diff
@@ -30,10 +31,13 @@ export interface Identity<T> {
  */
 export function timeout<T>(milliseconds: number, reason: Error): Identity<T> {
 	const start = Date.now();
-	return function (value: T): Promise<T> {
+	return function (value?: IdentityValue<T>): Promise<T> {
 		if (Date.now() - milliseconds > start) {
 			return Promise.reject<T>(reason);
 		}
+		if (typeof value === 'function') {
```

Comment:
Is this new functionality?

Reply:
Doesn't look like it's covered in tests

Reply:
It's not new functionality (`value` was `T | (() => T | Thenable<T>` before my changes), but it wasn't handled properly before. You're right that we need to test it.

Reply:
Tests have been added